src/block-components/typography/style.js

```diff
@@ -0,0 +1,204 @@
+					return undefined
+				}
+				return value
+			},
+			dependencies: [ 'textColorType' ],
+		},
+		{
+			selector,
+			styleRule: 'backgroundImage',
+			hover: 'all',
+			hoverSelector,
+			attrName: 'textColor1',
+			valuePreCallback: ( value, getAttribute, device, state ) => {
+				if (
+					getAttribute( 'textColorType', 'desktop', state ) !== 'gradient' ||
+					getAttribute( 'textColor1', 'desktop', state ) === '' ||
+					getAttribute( 'textColor2', 'desktop', state ) === ''
+				) {
+					return undefined
+				}
+				return value
+			},
+			valueCallback: ( value, getAttribute, device, state ) => {
+				const textColor1 = getAttribute( 'textColor1', 'desktop', state )
+				const textColor2 = getAttribute( 'textColor2', 'desktop', state )
+				const textGradientDirection = getAttribute( 'textGradientDirection', 'desktop', state )
+
+				return `linear-gradient(${ textGradientDirection !== '' ? `${ textGradientDirection }deg, ` : '' }${ textColor1 }, ${ textColor2 })`
+			},
+			dependencies: [ 'textColorType', 'textColor1', 'textColor2', 'textGradientDirection' ],
+		},
+		{
+			selector,
+			styleRule: 'textAlign',
+			attrName: 'textAlign',
+			hover: 'all',
+			hoverSelector,
+			responsive: 'all',
+		},
+		{
+			selector,
+			styleRule: 'lineHeight',
+			attrName: 'lineHeight',
+			hover: 'all',
+			hoverSelector,
+			responsive: 'all',
+			hasUnits: 'em',
+		},
+		{
+			selector,
+			styles: {
+				fontWeight: 'fontWeight',
+				textTransform: 'textTransform',
+			},
+			hover: 'all',
+			hoverSelector,
+		},
+		{
+			selector,
+			styleRule: 'fontFamily',
+			attrName: 'fontFamily',
+			valueCallback: value => getFontFamily( value ),
+		},
+		{
+			selector,
+			styleRule: 'letterSpacing',
+			attrName: 'letterSpacing',
+			hasUnits: 'px',
+			hover: 'all',
+		},
+	]
+}
+
+export const Style = props => {
+	const {
+		attributes,
+		options = {},
+		...propsToPass
+	} = props
+
+	const {
+		selector = '',
+		inherit = true, // If false, desktop styles will only be applied to desktop, etc.
+		inheritMax = 50, // If provided & inherit is true, clamp the inherited value in tablet and mobile to this.
+		inheritMin,
+		hoverSelector = '',
+	} = options
+
+	const styles = useStyles( attributes, getStyleParams( {
+		selector, inherit, inheritMin, inheritMax, hoverSelector,
+	} ) )
+
+	return (
+		<StyleComponent
+			styles={ styles }
+			versionAdded="3.0.0"
+			versionDeprecated=""
+			{ ...propsToPass }
+		/>
+	)
+}
+
+Style.Content = props => {
+	const {
+		attributes,
+		options = {},
+		...propsToPass
+	} = props
+
+	const {
+		selector = '',
+		inherit = true, // If false, desktop styles will only be applied to desktop, etc.
+		inheritMax = 50, // If provided & inherit is true, clamp the inherited value in tablet and mobile to this.
+		inheritMin,
+		hoverSelector = '',
+	} = options
+
+	const styles = getStyles( attributes, getStyleParams( {
+		selector, inherit, inheritMin, inheritMax, hoverSelector,
+	} ) )
```

Comment:
Pass `options` directly, then deconstruct inside the `getStyleParams` function instead
```suggestion
	const styles = getStyles( attributes, getStyleParams( options ) )
```